enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,340 @@
+  - "@csrwng" # for HyperShift changes
+  - "@spadgett" # for Console changes
+  - "@abutcher" # for CCO changes
+approvers:
+  - "@sdodson"
+  - "@joelanford"
+  - "@bparees"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "@JoelSpeed " # for CredentialsRequest changes
+creation-date: 2023-01-21
+last-updated: 2023-04-19
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OCPBU-4
+  - https://issues.redhat.com/browse/PORTENABLE-471
+  - https://issues.redhat.com/browse/CCO-286
+see-also:
+  - []
+replaces:
+  - []
+superseded-by:
+  - []
+---
+
+# Tokenized Authentication for Red Hat Operators on Cloud Providers
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Many Cloud Providers offer services that allow authentication via Temporary Access Tokens (TAT).
+
+For example, Amazon Web Services (AWS) Secure Token Service (STS) allows you to request temporary limited-privilege
+credentials. A concrete example is write permission on an S3 bucket on AWS. STS is supported on all the major cloud
+providers where OpenShift is deployed.
+
+Currently, several Red Hat operators provide detailed instructions for how to use the operator on clusters where CCO is
+in Manual Mode. The process is designed to have manual operations performed by the customer that are inconsistent across
+layered product operators.
+
+This enhancement seeks to bring a unified set of standards and tooling to the process of enabling an
+operator to access cloud provider resources using TAT, and improve the UX for consumers of these operators. 
+
+## Motivation
+
+Several Red Hat operators have normal operating modes where they assure their operands have required cloud 
+resources to perform properly. In Mint Mode, CCO will automatically create the IAM roles and credentials required for 
+the operator to assume the role and authenticate with the Cloud Provider.
+
+In Manual mode or more generally on clusters that support TAT (CCO not necessarily installed), an admin installing an 
+operator is required to:
+
+Extract the CredentialsRequest from the operator's image or codebase in order to know what IAM role is appropriate for 
+the operator to assume
+if using the [Cloud Credential Operator Utility](https://github.com/openshift/cloud-credential-operator/blob/master/docs/ccoctl.md), `ccoctl`:
+1. Find and download a compatible `ccoctl` binary
+2. Accept that `ccoctl` is a binary you downloaded that has god-mode access to your Cloud Provider
```

Comment:
Philosophically doesn't `ccoctl` need only as limited a credential set as any proposed automation for the CCO iteself? Is it not possible for a user to de-escalate `ccoctl` or just uncommon?

Reply:
Correct, in manual mode CCO doesn't actually have the ability to create the roles. Off cluster the IAM admin (not necessarily the cluster admin) runs `ccoctl`, which talks directly to the cloud, using the IAM admin's credentials to create roles.

CCO may in the future grow the ability to validate roles match the CredentialsRequest but will likely never be capable of role creation because most organizations are unwilling to entrust either the cluster or even the openshift admin to manage IAM.